### ISSUE TITLE
[SQLLab] A query with keyword instead of a literal runs into "Only `SELECT` statements are allowed against this database"

### ISSUE THREAD

**Issue by mrshu** [apache/incubator-superset#6998]
When trying to run the following query
```sql
WITH final AS (
  SELECT * 
  FROM b
)
SELECT * FROM final
```
 in Superset's SQL Lab one runs into the _"Only `SELECT` statements are allowed against this database"_ security exception, which is triggered on this line: https://github.com/apache/incubator-superset/blob/master/superset/sql_lab.py#L156

At first, I though that there was a bug with Superset's dealing with `WITH` queries, but it turns out that the reason is much more straightforward: since `sqlparse` considers `final` to be a keyword (as you can see [here](https://github.com/andialbrecht/sqlparse/blob/master/sqlparse/keywords.py#L264)), this whole query won't get properly persed, calling [`get_type()`](https://github.com/apache/incubator-superset/blob/master/superset/sql_parse.py#L55) will return `UNKNOWN`, the ["pessimistic readonly"](https://github.com/apache/incubator-superset/blob/master/superset/sql_parse.py#L60) will return `False` and the [security exception](https://github.com/apache/incubator-superset/blob/master/superset/sql_lab.py#L156) will be triggered.

I am not exactly sure what would be the proper way of dealing with a situation like this, as `final` is not really considered in Presto and the query from above would execute properly and I am pretty sure there are some good reasons why `sqlparse` considers `final` a keyword. I am therefore opening this issue in hopes that it may help save some debugging efforts in the future.

That being said, since we know that this happens when the `get_type()` call returns `UNKNOWN`, would it make sense to raise a different exception than the standard _"Only `SELECT` statements are allowed against this database"_? It may provide the user with more information as to how could they alter the query to around this problem -- currently the only recourse seems to be setting "Allow DML" for the connection.

If you do not find this issue actionable, please feel free to close it -- I am totally OK with that.

Thanks!

------------------------------------------------------

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Superset version
0.28

**Comment by mrshu** [comment]
@kristw @mistercrunch Would you guys be interested in a PR that would raise a different exception when the `get_type()` function call returns `UNKNOWN`? I'd be happy to put it together.

Thanks!

**Comment by jhua-wish** [comment]
This issue still exists in the latest superset. I'm suggesting to bump `sqlparse` package to the newest instead of `0.3.0`

**Comment by raags** [comment]
I stumbled into the same issue - would suggest we keep this issue open. And an interim solution could be a better error message?

**Comment by pacholoamit** [comment]
Any updates on this?

**Comment by rumbin** [comment]
BTW, `source` named CTEs will also cause this error.

**Comment by coolinstarxckr** [comment]
I ran into the same issue......any updates on this?

**Comment by rumbin** [comment]
Adding `sha` to the list of evil words, just for documentation

**Comment by GuruprasadaShridharHegde** [comment]
> When trying to run the following query
> 
> ```sql
> WITH final AS (
>   SELECT * 
>   FROM b
> )
> SELECT * FROM final
> ```
> 
> in Superset's SQL Lab one runs into the _"Only `SELECT` statements are allowed against this database"_ security exception, which is triggered on this line: https://github.com/apache/incubator-superset/blob/master/superset/sql_lab.py#L156
> 
> At first, I though that there was a bug with Superset's dealing with `WITH` queries, but it turns out that the reason is much more straightforward: since `sqlparse` considers `final` to be a keyword (as you can see [here](https://github.com/andialbrecht/sqlparse/blob/master/sqlparse/keywords.py#L264)), this whole query won't get properly persed, calling [`get_type()`](https://github.com/apache/incubator-superset/blob/master/superset/sql_parse.py#L55) will return `UNKNOWN`, the ["pessimistic readonly"](https://github.com/apache/incubator-superset/blob/master/superset/sql_parse.py#L60) will return `False` and the [security exception](https://github.com/apache/incubator-superset/blob/master/superset/sql_lab.py#L156) will be triggered.
> 
> I am not exactly sure what would be the proper way of dealing with a situation like this, as `final` is not really considered in Presto and the query from above would execute properly and I am pretty sure there are some good reasons why `sqlparse` considers `final` a keyword. I am therefore opening this issue in hopes that it may help save some debugging efforts in the future.
> 
> That being said, since we know that this happens when the `get_type()` call returns `UNKNOWN`, would it make sense to raise a different exception than the standard _"Only `SELECT` statements are allowed against this database"_? It may provide the user with more information as to how could they alter the query to around this problem -- currently the only recourse seems to be setting "Allow DML" for the connection.
> 
> If you do not find this issue actionable, please feel free to close it -- I am totally OK with that.
> 
> Thanks!
> 
> * [x]  I have checked the superset logs for python stacktraces and included it here as text if there are any.
> * [x]  I have reproduced the issue with at least the latest released version of superset.
> * [x]  I have checked the issue tracker for the same issue and I haven't found one similar.
> 
> ### Superset version
> 0.28

This is what worked for me:

I was using final in my sql which is a reserved keyword, after that i enclosed with double quotes like "final" which worked for me.. Make sure whenever you refer to the RESERVED keyword then enclose with quotes.. Thanks All !!

**Comment by GuruprasadaShridharHegde** [comment]
This is what worked for me:

I was using final in my sql which is a reserved keyword, after that i enclosed with double quotes like "final" which worked for me.. Make sure whenever you refer to the RESERVED keyword then enclose with quotes.. Thanks All !!

**Comment by GuruprasadaShridharHegde** [comment]
This is what worked for me:

I was using **final** in my sql which is a reserved keyword, after that i enclosed with double quotes like "final" which worked for me.. Make sure whenever you refer to the RESERVED keyword then enclose with quotes.. Thanks All !!
```
  "final" AS (
    SELECT
      vw.*,
      TRIM(gc.first_name||' '||gc.last_name) AS "influencer_name"
    FROM "vw" AS vw
```

**Comment by tgray017** [comment]
I ran into the same issue when I named my CTE `events`. Changing the name fixed it.

**Comment by fbgeobit** [comment]
It seems fixed on v 6.1.0, queries like this:

```sql
WITH 
final AS (
    SELECT
      'my awesome data' AS im_the_best
)
SELECT * FROM final
```

work as expected.
